tests/units/test_rp_agent.py

```diff
@@ -427,3 +538,298 @@ def test_analytics(mock_send_event, config):
+    ]
+    ba = BehaveAgent(config, mock_rps)
+    ba._feature_id = "feature_id"
+    ba._log_feature_cleanup(mock_context)
+    mock_rps.start_test_item.assert_has_calls(
+        [
+            mock.call(
+                name="Execution of 'cleanup_func1' cleanup function",
+                start_time=123,
+                item_type="AFTER_SUITE",
+                parent_item_id="feature_id",
+            ),
+            mock.call(
+                name="Execution of 'cleanup_func2' cleanup function",
+                start_time=123,
+                item_type="AFTER_SUITE",
+                parent_item_id="feature_id",
+            ),
+        ]
+    )
+    assert mock_rps.finish_test_item.call_count == 2
+
+
+def test_log_scenario_cleanup_no_scenario_layer():
+    cfg = Config(
+        endpoint="endpoint",
+        token="token",
+        project="project",
+        step_based="True",
+    )
+    mock_rps = mock.create_autospec(ReportPortalService)
+    mock_context = mock.Mock()
+    mock_context._stack = [{"@layer": "feature"}]
+    BehaveAgent(cfg, mock_rps)._log_scenario_cleanup(mock_context)
+    mock_rps.start_test_item.assert_not_called()
+
+
+def test_log_scenario_cleanup_no_cleanups():
+    cfg = Config(
+        endpoint="endpoint",
+        token="token",
+        project="project",
+        step_based="True",
+    )
+    mock_rps = mock.create_autospec(ReportPortalService)
+    mock_context = mock.Mock()
+    mock_context._stack = [{"@layer": "scenario"}]
+    BehaveAgent(cfg, mock_rps)._log_scenario_cleanup(mock_context)
+    mock_rps.start_test_item.assert_not_called()
+
+
+@mock.patch("behave_reportportal.behave_agent.timestamp")
+def test_log_scenario_cleanup(mock_timestamp):
+    cfg = Config(
+        endpoint="endpoint",
+        token="token",
+        project="project",
+        step_based="True",
+    )
+    mock_timestamp.return_value = 123
+    mock_rps = mock.create_autospec(ReportPortalService)
+    mock_context, mock_func1, mock_func2 = mock.Mock(), mock.Mock, mock.Mock()
+    mock_func1.__name__ = "cleanup_func1"
+    mock_func2.__name__ = "cleanup_func2"
+    mock_context._stack = [
+        {"@layer": "scenario", "@cleanups": [mock_func1, mock_func2]}
+    ]
+    ba = BehaveAgent(cfg, mock_rps)
+    ba._scenario_id = "scenario_id"
+    ba._log_scenario_cleanup(mock_context)
+    mock_rps.start_test_item.assert_has_calls(
+        [
+            mock.call(
+                name="Execution of 'cleanup_func1' cleanup function",
+                start_time=123,
+                item_type="AFTER_TEST",
+                parent_item_id="scenario_id",
+            ),
+            mock.call(
+                name="Execution of 'cleanup_func2' cleanup function",
+                start_time=123,
+                item_type="AFTER_TEST",
+                parent_item_id="scenario_id",
+            ),
+        ]
+    )
+    assert mock_rps.finish_test_item.call_count == 2
```

Comment:
Make it softly.

Reply:
Why?

Reply:
In case of raise condition, the next checks won't be run.

Reply:
It was the last verification in scope of test. Anyway mock assert method is used in this test and it's not soft

Reply:
1. Mock assert can be used with delayed_assert;
2. I see assert_has_calls call after this assert.
![image](https://user-images.githubusercontent.com/8349078/112290509-5bef5680-8ca0-11eb-9143-b3bc5988a9b8.png)

Reply:
This part of code was updated. It's last assert for test